pilot/test/mock/config.go

```diff
@@ -489,6 +489,7 @@ func CheckCacheEvents(store model.ConfigStore, cache model.ConfigStoreCache, nam
 	defer close(stop)
 
 	lock := sync.Mutex{}
+	cond := sync.NewCond(&lock)
 
 	added, deleted := 0, 0
```

Comment:
can you make added and deleted channels as well.
You will not need to use locks and conditions in that case.

`add_ch <- true`

Reply:
Channel cannot help with removing locks here unless we are willing to remove checks in 
switch ev {
		case model.EventAdd:
			if deleted != 0 {
				t.Errorf("Events are not serialized (add)")
			}

For this to work, we still have to keep the result out of channel for read asynchronously. 

better ideas?